### PR TITLE
support for getting data in blocks of custom size

### PR DESCRIPTION
**Pull request by srutar**
block reading -- again, the downloaded repo did not work for me and vs2012 changed a lot of packages, so consider merging only the .cs files

### CODE REVIEW

File: Controllers/ApiController.cs [hunk 1]
```diff
@@ -55,6 +55,25 @@ public object Get(string bin, string fileName)
...
 
+        public object GetBlock(string bin, string fileName, int blockN, int blockSize, string prevHash)
+        {
+            try
+            {
+                string hash;
+                string data = NestCore.Get(ServiceConfig.DataFolder, bin, fileName, blockN, blockSize, prevHash, out hash);
+                if (hash != null)
+                {
+                    string appendix = "\n\t" + hash;
```
Comment (mgrcar) [inline Controllers/ApiController.cs]: I think there should be Environment.NewLine in here somewhere :)

File: Controllers/ApiController.cs [hunk 2]
```diff
@@ -47,7 +47,44 @@ public object Get(string bin, string fileName)
...
+                }
+                return Content(data, "text/plain", Encoding.UTF8);
+            }
+            catch (Exception e)
+            {
+                return Error(e);
+            }
+        }
+
+        public object GetBlock(string bin, string fileName, int blockN, int blockSize, string prevHash)
```
Comment (mgrcar) [inline Controllers/ApiController.cs]: This function is duplicated. I assume this one is obsolete, but I will let you revise this. How did this happen - did U get merge something?

**Comment by srutar** [comment]
Funny, of course it's unintentional duplicate... remove :-) tnx

On December 21, 2016 8:57:10 AM GMT+01:00, "Miha Grčar" <notifications@github.com> wrote:
>mgrcar requested changes on this pull request.
>
>Please check that one issue... :)
>
>> +            {
>+                string hash;
>+                string data = NestCore.Get(ServiceConfig.DataFolder,
>bin, fileName, blockN, blockSize, prevHash, out hash);
>+                if (hash != null)
>+                {
>+                   
>HttpContext.Response.AppendHeader(NestUtils.ReadBlockHashHeaderName,
>hash);
>+                }
>+                return Content(data, "text/plain", Encoding.UTF8);
>+            }
>+            catch (Exception e)
>+            {
>+                return Error(e);
>+            }
>+        }
>+
>+        public object GetBlock(string bin, string fileName, int
>blockN, int blockSize, string prevHash)
>
>This function is duplicated. I assume this one is obsolete, but I will
>let you revise this. How did this happen - did U get merge something?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/SowaLabs/NestService/pull/1#pullrequestreview-13914135

-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Comment by mgrcar** [comment]
Hey,

Sel sem cez tvojo kodo - looks great :) Sem zmergal ...

BTW, NestCore, Client, Service so zdaj en sam repo, Nest. Zato z diska 
odstrani folderje Nest*, kloniraj novi Nest, popravi reference v svojih 
projektih ...

Thx, LP,
mIHA

**Pull request merged**